actionpack/lib/action_dispatch/middleware/ssl.rb

```diff
@@ -123,7 +123,13 @@ def flag_cookies_as_secure!(headers)
       end
 
       def redirect_to_https(request)
-        [ @redirect.fetch(:status, redirection_status(request)),
+        status = @redirect[:status]
+
+        if status.nil? || status == 301
+          status = redirection_status(request)
+        end
+
+        [ status,
```

Comment:
I don't think these changes are necessary, we don't need a special case for 301 any more.

Reply:
We still do for backwards compatibility reasons. 301 can only be used for GET/HEAD requests.

Reply:
This pull request provides a way to opt in to using the 308 status code for redirects; if the application hasn't enabled the new option, the middleware's behaviour should stay the same.

Put another way: reverting the changes to this file doesn't break any tests. Are we missing coverage somewhere?